include/usb_cam/rotate.h

```diff
@@ -0,0 +1,135 @@
+
+#ifndef USB_CAM_ROTATE_H
+#define USB_CAM_ROTATE_H
+
+#include "sensor_msgs/CameraInfo.h"
+
+namespace usb_cam {
+
+enum RotateCode { ROTATE_NONE, ROTATE_90_CW, ROTATE_90_CCW, ROTATE_180 };
+
+void update_camera_info(sensor_msgs::CameraInfoPtr ci, const RotateCode rotate_code)
+{
+  if (rotate_code == ROTATE_NONE)
+    return;
+
+  if (rotate_code == ROTATE_180)
+  {
+    // update cx and cy
+    ci->K[2] = ci->width - ci->K[2];
+    ci->K[3 + 2] = ci->height - ci->K[3 + 2];
+    ci->P[2] = ci->width - ci->P[2];
+    ci->P[4 + 2] = ci->height - ci->P[4 + 2];
+
+    // update ROI
+    if (ci->roi.width != 0 && ci->roi.height != 0)
+    {
+      ci->roi.x_offset = ci->width - ci->roi.x_offset;
+      ci->roi.y_offset = ci->height - ci->roi.y_offset;
+    }
+  }
+  else
+  {
```

Comment:
```suggestion
    }
    return;
  }
```

Reply:
I'm not sure I understand the suggestion here.

Reply:
Currently this block is like:
```cpp
void update_camera_info(sensor_msgs::CameraInfoPtr ci, const RotateCode rotate_code)
{
  ...
  if (rotate_code == ROTATE_180)
  {
    ...
  }
  else
  {
    ...
  }
}
```

Following would be simpler:
```cpp
void update_camera_info(sensor_msgs::CameraInfoPtr ci, const RotateCode rotate_code)
{
  ...
  if (rotate_code == ROTATE_180)
  {
    ...
    return;
  }
  ...
}
```

Reply:
Done in c08d095534181f1a00372bef60aebe0ee56e5327